coreax/score_matching.py

```diff
@@ -521,7 +540,7 @@ def score_function(x_: ArrayLike) -> Array:
             """
             # Check format
             original_number_of_dimensions = jnp.asarray(x_).ndim
-            x_ = jnp.atleast_2d(x_)
+            x_ = _atleast_2d_consistent(x_)
```

Comment:
This single line change appears to be causing the performance issues.

Reply:
Hmm, that is interesting, will look into why this is slower

Reply:
I have removed some of the code that ended up not being need in `_atleast_2d_consistent`, hopefully that speeds things up

Reply:
If this doesn't fix things, then I assume there is some weird dimension issue going on in the score matching code, which I wouldn't be surprised by. Will try track down if that is the case

Reply:
Having investigated this more closely, it seems that the particular issue is caused by us adding the new dimension in axis 1 rather than axis 0.

Replacing this line with
```python
            if len(x_.shape) == 1:
                x_ = jnp.expand_dims(x_, 1)
            x_ = jnp.array(x_, copy=False, ndmin=2)
```
(i.e. the internals of `__atleast_2d_consistent`) yields no performance improvement. However, using 
```python
            if len(x_.shape) == 1:
                x_ = jnp.expand_dims(x_, 0)
            x_ = jnp.array(x_, copy=False, ndmin=2)
```
(the same code, just adding the dimension in a different spot) instead yields the original performance.